### ISSUE TITLE
handlebars targeting last 'title' created instead of current title

### ISSUE THREAD

**Issue by RpRoderick** [ga-wdi-boston/group-project#226]
I have a function written to create the hash of our url in handlebars.  I need it to target the 'title' of the clicked on button and it targets the 'title' of the LAST survey created instead.  I thought that @index would be used for this, but it does not work.  How can I correctly target the current title using handlebars?  All combinations of 'this' (that I've tried) haven't worked.  

```
{{#each surveys}}
  <div class="imageImage">

      <button class="btn btn-warning" id="abc"> <h4>{{title}}</h4></button>

    <script>
    jQuery(function ($) {
        $('#abc').on('click', function () {
          location.hash = `{{title}}`.replace(' ','')               <----this {{title}} is where I can't get it to work
        // location.hash = Math.random().toString();
        });
    });
    </script>
  </div>
{{/each}}
```

**Comment by gaand** [comment]
An `id` must be unique in the document.  You are using the same `id` for each button.

**Comment by RpRoderick** [comment]
I changed the click handler to target the class of button instead (which I named svy-btn) but the events still had trouble assigning the correct data into the fields.  It ended up being that handlebars assigned the last created title to every button no matter what the click event targeted.  The event would iterate through the surveys and each survey would get assigned the same data.  I ended up hard coding the event to pass the data after on document ready but within the 'body' so that the data would not be assigned until that point.  Using the attribute data-id to assign {{_id}}, the location.hash could then target the specific id of each individual button instead of iterating through all of them.  Again, I thought @index would have worked for this directly in handlebars?  But this was a quick work around, located within addHandlers, which loads on Doc Ready.

```
$('body').on('click','.svy-btn', function(){
    location.hash = $(this).attr('data-id');
  });
```
